### PR TITLE
Added fallback to xterm if gnome-terminal doesn't exist

### PR DESCRIPTION
**Pull request by pieandcakes**
@jacdavis @gregg-miskelly This pull request is to allow the terminal fallback to xterm if gnome-terminal isn't available

### CODE REVIEW

File: src/MICore/Transports/LocalLinuxTransport.cs [hunk 1]
```diff
@@ -110,6 +111,19 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
...
             bool requiresRootAttach = this.GetPtraceScope() != 0;
 
+            // Check and see if gnome-terminal exists. If not, fall back to xterm
+            string terminalCmd, bashCommandPrefix;
+            if (File.Exists(GnomeTerminalPath))
+            {
+                terminalCmd = GnomeTerminalPath;
+                bashCommandPrefix = "--title DebuggerTerminal -x";
+            }
+            else
```
Comment (gregg-miskelly) [inline src/MICore/Transports/LocalLinuxTransport.cs]: Should we have some failure message if both are unavailable?

**Comment by gregg-miskelly** [comment]
Looks good aside from comment.

File: src/MICore/Transports/LocalLinuxTransport.cs [hunk 2]
```diff
@@ -144,15 +158,16 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
                 }
             }
 
-            string argumentString = string.Format(System.Globalization.CultureInfo.InvariantCulture,
-                    "--title DebuggerTerminal -x bash -c \"cd {0}; DbgTerm=`tty`; trap 'rm {2}; rm {3}' EXIT; {1} --interpreter=mi --tty=$DbgTerm < {2} > {3};\"",
+            string argumentString = string.Format(CultureInfo.InvariantCulture,
+                    "{4} bash -c \"cd {0}; DbgTerm=`tty`; trap 'rm {2}; rm {3}' EXIT; {1} --interpreter=mi --tty=$DbgTerm < {2} > {3};\"",
```
Comment (jacdavis) [inline src/MICore/Transports/LocalLinuxTransport.cs]: I would normally nitpick the order of the params, but given how complex this is (and the fact that I too am guilty of this) I shall keep my mouth shut.

**Comment by jacdavis** [comment]
LGTM. Let me know when you're ready for us to merge.

File: src/MICore/Transports/LocalLinuxTransport.cs (hunk 1 above)
Comment (pieandcakes) [inline src/MICore/Transports/LocalLinuxTransport.cs]: Probably, but right now it just says that gnome-terminal (or xterm) isn't available

**Pull request merged**